NMS-14172: Clean up ticketer docs (#4539)
OpenNMS/opennms: docs/modules/reference/pages/ticketing/ticketer/jira.adoc

[[ref-ticketing-jira-setup]]
== Setup

IMPORTANT: The use of the JIRA ticketer requires the installation of the `opennms-plugin-ticketer-jira` package before the plugin will start.

****
To install the opennms-plugin-ticketer-jira package:

* For RPM-based distributions:
** `yum install opennms-plugin-ticketer-jira`
* For Debian-based distributions:
** `apt-get install opennms-plugin-ticketer-jira`
****

. Once installed, set the following property in `$OPENNMS_HOME/etc/opennms.properties.d/ticketer.properties` to enable the plugin:

 opennms.ticketer.plugin=org.opennms.netmgt.ticketd.OSGiBasedTicketerPlugin

. Configure the plugin options by setting desired options in `$OPENNMS_HOME/etc/jira.properties`.
See the table below for available options.

> . To make the ticketer system start automatically in the future, add `jira-troubleticketer` to the `featuresBoot` property in the `$OPENNMS_HOME/etc/org.apache.karaf.features.cfg`.

Reviewer: echo "jira-troubleticketer" >> $OPENNMS_HOME/etc/featuresBoot.d/ticketer.boot

Bad practice to append things to org.apache.karaf.features.cfg, as it can be overwritten by fix-karaf-setup.sh, or by upgrade.

[[ref-ticketing-jira-setup]]
== Setup

IMPORTANT: The use of the JIRA ticketer requires the installation of the `opennms-plugin-ticketer-jira` package before the plugin will start.

****
To install the opennms-plugin-ticketer-jira package:

* For RPM-based distributions:
** `yum install opennms-plugin-ticketer-jira`
* For Debian-based distributions:
** `apt-get install opennms-plugin-ticketer-jira`
****

> . Once installed, set the following property in `$OPENNMS_HOME/etc/opennms.properties.d/ticketer.properties` to enable the plugin:

Reviewer 2: ```suggestion
. Once installed, set the following property in `$OPENNMS_HOME/etc/opennms.properties.d/ticketer.properties` to enable the plugin (create the file if it does not exist):
```

> (Create the file if it does not exist)

Reviewer 2: ```suggestion

```

 opennms.ticketer.plugin=org.opennms.netmgt.ticketd.OSGiBasedTicketerPlugin

. Configure the plugin options by setting desired options in `$OPENNMS_HOME/etc/jira.properties`.
See the table below for available options.
. To make the ticketer system start automatically on future service starts, add the line `jira-troubleticketer` to the file `$\{OPENNMS_HOME}/etc/featuresBoot.d/ticketer.boot`.

> (Create the file if it does not exist)

Reviewer 2: ```suggestion
(Create the file if it does not exist.)
```

. To make the ticketer system start now, log in to the Karaf shell and run the following command to load the feature:

 feature:install jira-troubleticketer

The plugin should be ready to use.

> Login to the Karaf shell and run `opennms:jira-verify` to check your configuration.

Reviewer 2: ```suggestion
Log in to the Karaf shell and run `opennms:jira-verify` to check your configuration.
```

> .`jira.properties` Options

Reviewer 2: ```suggestion
.jira.properties Options
```
removing formatting here, as it makes it look strange in the table label.

|===

NOTE: The transition names for `resolve` and `reopen` are typically found on buttons when looking at the ticket in Jira.

NOTE: Use either the `opennms:jira-list-issue-types` Karaf Command or https://confluence.atlassian.com/display/JIRA050/Finding+the+Id+for+Issue+Types[Atlassian's knowledgebase] to determine the appropriate issue type ID.

== Jira commands

The Jira ticketing plugin provides Karaf commands to help set up the plugin on the Karaf shell.
There are Karaf commands to list all available projects, versions, components, groups, issue types, and more.
To list all available commands simply type `help | grep jira` in the Karaf shell.
Include `--help` after a command to see details on a specific command.

== Custom fields

Besides the common fields creator, create date, description, or subject, you also may also need to set ticket-system proprietary fields.
The Jira ticketing plugin provides the opportunity to define those in the OpenNMS ticket attributes, which can be overwritten with Drools rules.

To enable the Drools ticketing integration, set the following property in `$OPENNMS_HOME/etc/opennms.properties.d/ticketer.properties`:

> (Create the file if it does not exist)

Reviewer 2: ```suggestion
(Create the file if it does not exist.)
```
